### DISCUSSION TITLE
Utility functions

### DISCUSSION THREAD

**Discussion by ejamshidiasl** [KilledByAPixel/LittleJS#107] [Ideas]
i think we need some usefull functions (like godot). maybe these things:

- look at point
- move to point
- tween variable
- is on floor?
- get input value by axis and vector
- zoom, pan camera with mouse and touch
- camera follow target
- etc.

**Comment by swkidd** [comment]
@ejamshidiasl Fyi, for 'is on floor?' there is an engine variable called `groundObject` that can be used to test if an `EngineObject` is on the ground

**Comment by KilledByAPixel** [comment]
Thank you for the suggestions! i can see how some of these would be useful. There isn't any camera logic at all in the engine, maybe some stuff for that would be helpful. part of the goal of LJS is to keep things simple to empower users to write their own functions for this kind of thing like how the camera works in their game because there are so many different types of cameras. I will think about what I can do to help make it easier to do this stuff tough. Keep adding to the list if there is anything else you feel is missing!

thanks @swkidd! LJS uses a streamlined approach so there isn't a function like "isOnGround()". There is just a variable on objects called groundObject which is truthy if an object is on the ground. Usually it is the object it is on top of but it may just be a true value if it is on the the ground on something that is not a real object like the tile collision.

**Comment by KilledByAPixel** [comment]
This stuff seems all well supported now...

look at point - setCameraPos()
move to point - setCameraPos()
tween variable - tween system is in progress
is on floor? - EngineObject.groundObject
get input value by axis and vector - not sure what this means
zoom, pan camera with mouse and touch - [Camera Mouse Drag Demo](https://killedbyapixel.github.io/LittleJS/examples/?example=Camera+Mouse+Drag)
camera follow target - in object's update setCameraPos(this.pos)
